### PR TITLE
fix: adapt tooltip message on Share button when there are unsaved changes

### PR DESCRIPTION
**Pull request by csm-thu**

### CODE REVIEW

File: public/locales/en/translation.json [hunk 1]
```diff
@@ -30,6 +30,9 @@
           }
         }
       },
+      "shareScenario": {
```
Comment (ardvi) [inline public/locales/en/translation.json:33]: it's weird to create a new entry here, all other labels are in commoncomponents.dialog.share.button

File: src/components/ShareCurrentScenarioButton/ShareCurrentScenarioButton.js [hunk 2]
```diff
@@ -22,6 +24,11 @@ const ShareCurrentScenarioButton = () => {
   } = useShareCurrentScenarioButton();
 
   const { isDirty } = useFormState();
+  if (isDirty && shareScenarioDialogLabels?.button?.tooltip)
```
Comment (ardvi) [inline src/components/ShareCurrentScenarioButton/ShareCurrentScenarioButton.js:27]: i think it's better to modify getShareScenarioDialogLabels for get the good label

**Review by ardvi — COMMENTED** [review]
Some comments

File: src/components/ShareCurrentScenarioButton/ShareCurrentScenarioButton.js (hunk 2 above)
Comment (esasova) [reply src/components/ShareCurrentScenarioButton/ShareCurrentScenarioButton.js:27]: Maybe, it's better to change it in ui package and add a disabled tooltip label there ? :slightly_smiling_face:  (or make like in CreateScenarioLabels in the function itself, yes)

**Review by nborde-CSM — APPROVED** [review]
LGTM :+1:

File: public/locales/en/translation.json (hunk 1 above)
Comment (csm-thu) [reply public/locales/en/translation.json]: Done :slightly_smiling_face:

File: src/components/ShareCurrentScenarioButton/ShareCurrentScenarioButton.js (hunk 2 above)
Comment (csm-thu) [reply src/components/ShareCurrentScenarioButton/ShareCurrentScenarioButton.js]: Done, thank you for the feedback :slightly_smiling_face: 
I moved the call to the `useFormState` hook in the `ShareCurrentScenarioButtonHook` to get the value of `isDirty`, and added it as a parameter of the `getShareScenarioDialogLabels` function.

Modifying the UI component would mean to add a new prop for the form `isDirty` state to choose between possible tooltips labels: it's probably best to keep this logic in the webapp repo and provide only the tooltip message to the generic component.

**Review by ardvi — APPROVED** [review]
LGTM :+1:

**Review by esasova — APPROVED** [review]
LGTM :+1:

**Pull request merged**